guidance/guidance_los/guidance_los/los_guidance_algorithm.py

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env python3
+"""Third-order Line-of-Sight (LOS) guidance algorithm with reference filtering.
+
+This module implements a Line-of-Sight guidance algorithm with third-order reference
+filtering for smooth motion control. The algorithm generates guidance commands
+(surge, pitch, yaw) based on current position and target position while ensuring
+smooth transitions through third-order filtering.
+
+Key features:
+- LOS guidance with look-ahead distance scaling
+- Third-order reference filtering for smooth motion
+- Adaptive speed control based on cross-track error
+- Configurable natural frequency and damping parameters
+"""
+
+from typing import Tuple
+
```

Comment:
Most of the typing types are deprecated, you can just use `tuple` directly, and remove this import